Documentation/Global/Be/Infobox.rst

```diff
@@ -24,27 +32,92 @@ user to underline the meaning of the information.
 Possible values are in range from ``-2`` to ``2``. Please choose a
 meaningful value from the following list.
 
-``-2``
-   Notices (Default)
-``-1``
-   Information
-``0``
-   Positive feedback
-``1``
-   Warnings
-``2``
-   Error
+..  tabs::
+
+    ..  group-tab:: Screenshot
+
+        ..  figure:: /Images/BeInfobox.png
+            :alt: A screenshot demonstrating all possible infobox colors and states.
+
+            A demonstration of all possible states
+
+    ..  group-tab:: Numeral values
+
+        ``-2``
+           Notices (Default)
+        ``-1``
+           Information
+        ``0``
+           Positive feedback
+        ``1``
+           Warnings
+        ``2``
+           Error
+
+    ..  group-tab:: Code example
+
+        It is considered best practice to use the states from
+        :php:`\TYPO3\CMS\Fluid\ViewHelpers\Be\InfoboxViewHelper` together with the
+        `Constant ViewHelper <f:constant> <https://docs.typo3.org/permalink/t3viewhelper:typo3fluid-fluid-constant>`_:
+
+        ..  literalinclude:: _Infobox/_States.html
+            :caption: EXT:my_extension/Resources/Private/Backend/Templates/MyModule.html
 
 ..  _typo3-fluid-be-infobox-example:
 
 Examples
 ========
 
-Simple::
+Info box of level notice with a static title and a static text:
+
+..  code-block:: html
+    :caption: EXT:my_extension/Resources/Private/Backend/Templates/MyModule.html
+
+    <f:be.infobox title="Message title">your box content</f:be.infobox>
 
-   <f:be.infobox title="Message title">your box content</f:be.infobox>
+Warning box with disabled icon:
 
-All options::
+..  code-block:: html
+    :caption: EXT:my_extension/Resources/Private/Backend/Templates/MyModule.html
 
-   <f:be.infobox title="Message title" message="your box content" state="{f:constant(name: 'TYPO3\CMS\Fluid\ViewHelpers\Be\InfoboxViewHelper::STATE_NOTICE')}" iconName="check" disableIcon="true" />
+    <f:be.infobox
+        title="Message title"
+        message="your box content"
+        state="{f:constant(name: 'TYPO3\CMS\Fluid\ViewHelpers\Be\InfoboxViewHelper::STATE_WARNING')}"
+        iconName="check"
```

Comment:
Interesting approach. Add an icon to a non required property and then disable the rendering of the icon explizit. No need to change, but it's a little bit funny ;-)

Reply:
didnt see that, removed